Mathlib/Probability/Moments.lean

```diff
@@ -333,4 +344,363 @@ theorem measure_le_le_exp_cgf [IsFiniteMeasure μ] (ε : ℝ) (ht : t ≤ 0)
+      filter_upwards [hb] with ω hb
+      using exp_le_exp.mpr (mul_le_mul_of_nonneg_left hb (le_of_lt ht))
+    have p : ∀ᵐ ω ∂μ, ‖rexp (t * X ω)‖₊ ≤ rexp (t * b) := by
+      filter_upwards [b'] with ω b'
+      rw [(by simp only [coe_nnnorm, norm_eq_abs, abs_exp] : ‖rexp (t * X ω)‖₊ = rexp (t * X ω))]
+      exact b'
+    have p'' : ∫⁻ ω, ‖rexp (t * X ω)‖₊ ∂μ ≤ ∫⁻ _, ‖rexp (t * b)‖₊ ∂μ := by
+      apply lintegral_mono_ae
+      filter_upwards [p] with ω p
+      simp only [ENNReal.coe_le_coe]
+      rw [← (by simp only [coe_nnnorm, norm_eq_abs, abs_exp] : ‖rexp (t * b)‖₊ = rexp (t * b))] at p
+      exact p
+    suffices ∫⁻ _, ↑‖rexp (t * b)‖₊ ∂μ < ⊤ from lt_of_le_of_lt p'' this
+    simp only [lintegral_const]
+    apply ENNReal.mul_lt_top ENNReal.coe_lt_top (IsFiniteMeasure.measure_univ_lt_top)
+  exact ⟨aestronglyMeasurable_iff_aemeasurable.mpr <|
+    measurable_exp.comp_aemeasurable' (hX.const_mul t), hm1⟩
+
+lemma integrable_expt_bound [IsFiniteMeasure μ] {X : Ω → ℝ} {t a b : ℝ} (hX : AEMeasurable X μ)
+    (h : ∀ᵐ ω ∂μ, X ω ∈ Set.Icc a b) :
+    Integrable (fun ω ↦ exp (t * (X ω))) μ := by
+  have ha : ∀ᵐ ω ∂μ, a ≤ X ω := h.mono fun ω h ↦ h.1
+  cases lt_trichotomy t 0 with
+  | inr ht => cases ht with
+    | inr ht => exact integrable_expt X t b ht hX (h.mono fun ω h ↦ h.2)
+    | inl ht => rw [ht]; simp only [zero_mul, exp_zero, integrable_const]
+  | inl ht =>
+    rw [(by ext ω; rw [(by ring : - t * (- X ω) = t * X ω)] :
+      (fun ω ↦ rexp (t * X ω)) = (fun ω ↦ rexp (- t * (- X ω))))]
+    exact integrable_expt _ _ _ (by linarith : -t > 0) (AEMeasurable.neg hX)
+      (by filter_upwards [ha] with ω ha using neg_le_neg ha)
+
+lemma tilt_var_bound [IsProbabilityMeasure μ] (a b t : ℝ) (X : Ω → ℝ)
+    (h : ∀ᵐ ω ∂μ, X ω ∈ Set.Icc a b)
+    (hX : AEMeasurable X μ) :
+    variance X (μ.tilted (fun ω ↦ t * X ω)) ≤ ((b - a) / 2) ^ 2 := by
+  have _ : IsProbabilityMeasure (μ.tilted fun ω ↦ t * X ω) :=
+    isProbabilityMeasure_tilted (integrable_expt_bound hX h)
+  exact variance_le_sq_of_bounded
+      ((tilted_absolutelyContinuous μ fun ω ↦ t * X ω) h)
+      (AEMeasurable.mono_ac hX (tilted_absolutelyContinuous μ fun ω ↦ t * X ω))
+
+theorem integrable_bounded [IsFiniteMeasure μ] (a b : ℝ)
+    (X : Ω → ℝ) (hX : AEMeasurable X μ) (h : ∀ᵐ ω ∂μ, X ω ∈ Set.Icc a b) :
+    Integrable X μ := by
+  apply MeasureTheory.memℒp_one_iff_integrable.mp
+  apply memℒp_of_bounded h (aestronglyMeasurable_iff_aemeasurable.mpr hX)
+
+/-- Derivation of `mgf X μ t` is `μ[exp (t * X ω) * X ω]`.
+In order to deal with the differentiation of parametric integrals,
+`hasDerivAt_integral_of_dominated_loc_of_deriv_le` are used in the proof. -/
+theorem tilt_first_deriv [IsFiniteMeasure μ] (t a b : ℝ)
+    (X : Ω → ℝ) (hX : AEMeasurable X μ) (h : ∀ᵐ ω ∂μ, X ω ∈ Set.Icc a b):
+    let g := fun t ↦ mgf X μ t
+    let g' := fun t ↦ μ[fun ω ↦ rexp (t * X ω) * X ω]
+    Integrable (fun ω ↦ rexp (t * X ω) * X ω) μ → HasDerivAt g (g' t) t := by
```

Comment:
That result should hold more generally than for bounded random variables. Can you generalize it?
Is `h` used only to get integrability properties, or in a more essential way?

Reply:
h is only used to satisfy the regular conditions of dominated convergence theorem, so If we include the regularity condition directly in the assumptions, it would result in a stronger lemma.

Reply:
I wrote the PR #20151 to prove more generally that the mgf is analytic on the interior of its domain (without any bounded support assumption), and give the values of its derivatives. See `hasDerivAt_mgf` in that PR.
I suggest we wait for my PR to be merged and then use it here.

Reply:
Makes sense! Waiting for #20151 to be merged would be the better approach here